File: drills/ts/harnesses/nc54_sum_of_two_integers.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.getSum !== "function") throw new Error("__signature_mismatch__");
  return [
    ["getSum(1, 2)", show(3), show(solution.getSum(1, 2))],
    ["getSum(2, 3)", show(5), show(solution.getSum(2, 3))],
    ["getSum(-1, 1)", show(0), show(solution.getSum(-1, 1))],
    ["getSum(-2, -3)", show(-5), show(solution.getSum(-2, -3))],
    ["getSum(0, 0)", show(0), show(solution.getSum(0, 0))],
    ["getSum(-1, -1)", show(-2), show(solution.getSum(-1, -1))],
    ["getSum(5, -3)", show(2), show(solution.getSum(5, -3))],
  ];
}

File: drills/go/harnesses/nc54_sum_of_two_integers.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("getSum(1, 2)", 3, getSum(1, 2)),
			tc("getSum(2, 3)", 5, getSum(2, 3)),
			tc("getSum(-1, 1)", 0, getSum(-1, 1)),
			tc("getSum(-2, -3)", -5, getSum(-2, -3)),
			tc("getSum(0, 0)", 0, getSum(0, 0)),
			tc("getSum(5, -3)", 2, getSum(5, -3)),
		}
	})
}
